heka/sandbox/encoders/es_fields.lua

```diff
@@ -169,6 +170,11 @@ function process_message()
         end
     end
 
+    local ok, payload = pcall(cjson.decode, read_message("Payload"))
+    if ok and payload.application.architecture then
```

Comment:
Check the type on payload.application.  If it is not a table this encoder will be terminated.